walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/OpenIdPresentationRequest.kt

```diff
@@ -39,7 +45,39 @@ internal class OpenIdPresentationRequest(
     // Completes the presentation request and returns Result with success status if successful.
     override suspend fun complete(): VerifiedIdResult<Unit> {
         return getResult {
-            OpenIdResponder.sendPresentationResponse(request.rawRequest, requirement)
+            if (libraryConfiguration.isPreviewFeatureEnabled(PreviewFeatureFlags.FEATURE_FLAG_PRESENTATION_EXCHANGE_SERIALIZATION_SUPPORT)) {
+                val builder = PresentationExchangeResponseBuilder(libraryConfiguration)
+                builder.serialize(requirement, StringVerifiedIdSerializer)
+                val vpTokens = builder.buildVpTokens(
+                    request.presentationRequest.content.clientId,
+                    request.presentationRequest.content.nonce)
+                val idToken = builder.buildIdToken(
+                    request.presentationRequest.getPresentationDefinitions().first().id,
+                    request.presentationRequest.content.clientId,
+                    request.presentationRequest.content.nonce,
+                )
+
+                val result = if (vpTokens.size > 1) {
+                    libraryConfiguration.httpAgentApiProvider.presentationApis.sendResponses(
+                        request.presentationRequest.content.redirectUrl,
+                        idToken,
+                        vpTokens,
+                        request.presentationRequest.content.state
```

Comment:
I remember additional headers were being added so that they can be configured. Is that part of another PR?

Reply:
I don't see that in any of the change paths here. We have the parameter on the internal branch but I think that was a legacy change while we were creating the extension framework. Custom headers can be added via HttpAgent injection, and extensions add prefer header values on load. Is there another scenario requiring header manipulation for OpenIdPresentation I'm missing?